Sources/SimplenoteSearch/Extensions/NSPredicate+Search.swift

```diff
@@ -5,31 +5,34 @@ import Foundation
 //
 extension NSPredicate {
 
-    /// Returns a collection of NSPredicates that will match, as a compound, a given Search Text
+    /// Returns a collection of NSPredicates that will match, as a compound, a given Search Query
     ///
-    @objc(predicateForSearchText:)
-    public static func predicateForNotes(searchText: String) -> NSPredicate {
-        let keywords = searchText.trimmingCharacters(in: .whitespacesAndNewlines).components(separatedBy: .whitespaces)
-        var output = [NSPredicate]()
-
-        for keyword in keywords where keyword.isEmpty == false {
-            guard let tag = keyword.lowercased().suffix(afterPrefix: lowercasedSearchOperatorForTags) else {
-                output.append( NSPredicate(format: "content CONTAINS[cd] %@", keyword) )
-                continue
+    @objc(predicateForSearchQuery:)
+    public static func predicateForNotes(query: SearchQuery) -> NSPredicate {
+        let predicates = query.items.compactMap { (item) -> NSPredicate? in
+            switch item {
+            case .keyword(let value):
+                return NSPredicate(format: "content CONTAINS[cd] %@", value)
+            case .tag(let value):
+                if value.isEmpty {
+                    return nil
+                }
```

Comment:
This check can be dropped by filtering out empty tags in `parse()`. Note added below!